number-46/index.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html>
+  <head>
+  </head>
+  <body>
+    <script>
+      class Shape {
+        constructor(length, width) {
+          this.length = length;
+          this.width = width;
+        }
+
+        get area() {
+          return this.length * this.width;
+        }
+
+        get name() {
+          return "Default Shape";
+        }
+      }
+
+      class Square extends Shape {
+        constructor(length) {
+          super(length, length);
+        }
+
+        get name() {
+          return "Square";
+        }
+      }
+
+      class Rectangle extends Shape {
+        constructor(length, width) {
+          super(length, width);
+        }
+
+        get name() {
+          return "Rectangle";
+        }
+      }
+
+      class Triangle extends Shape {
+        constructor(length, width) {
+          super(length, width / 2);
+        }
+
+        get name() {
+          return "Triangle";
+        }
+      }
+      class Circle extends Shape {
+        constructor(radius) {
+          super(radius * radius, Math.PI);
```

Comment:
This is actually an interesting one! Since a circle doesn't really have a length or height, but what you did is clever since it would still leverage the `area` function 👍. One thing to note is that it's not always necessary to call `super`.